### PR TITLE
Fix building OBS Studio on FreeBSD

### PR DESCRIPTION
**Pull request by obiwac**
### Description

Fix building OBS Studio on FreeBSD:
- `libobs/graphics/graphics.h`: The `gs_texture_create_from_dmabuf` & `gs_query_dmabuf_*` functions were marked as Linux-only as of f7a55f45fdacf9c8383cafae8170294ac8997954.
- `plugins/linux-v4l2/v4l2-output.c`: `versionsort` (and by extension, `__strverscmp`) are not available on FreeBSD, as they are GNU extensions (`versionsort` is used by `virtualcam_start` to read `/dev/video*` devices in the right order), so implemented them.
- `UI/platform-x11.cpp`: Was calling `to_string` on FreeBSD when not `using namespace std;`, so calling `std::to_string` instead.

### Motivation and Context

Fixes building on FreeBSD.

### How Has This Been Tested?

Tested on FreeBSD 13.0-RELEASE-p4 and FreeBSD-CURRENT, hardware is non-applicable.
Compiled with clang version 11.0.1 & 13.0.1.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

Perhaps this is usererror, but `git-clang-format` doesn't seem to work?

```
YAML:35:21: error: unknown key 'AfterExternBlock'
  AfterExternBlock: false
                    ^~~~~
Error reading /usr/home/obiwac/obs-studio/.clang-format: Invalid argument
error: `clang-format -lines=178:178 -lines=187:187 -lines=191:191 -lines=196:196 -lines=198:255 -lines=258:258 -lines=260:260 plugins/linux-v4l2/v4l2-output.c` failed
```

So I attempted to format things myself.
Hope that's alright :)

### CODE REVIEW

**Comment by kkartaltepe** [comment]
I don't think we will want to include an implementation of versionsort from glibc just for freebsd. Is there an alternative function that can be used on freebsd?

**Comment by obiwac** [comment]
> I don't think we will want to include an implementation of versionsort from glibc just for freebsd. Is there an alternative function that can be used on freebsd?

Yeah, it's not ideal indeed, but again, to my knowledge and after a good look, I don't think there is.
There does exist `alphasort` though, as can be seen in the [`scandir(3)`](https://www.freebsd.org/cgi/man.cgi?query=scandir&sektion=3&format=html) manual, but that means differing behaviour on different platforms.

Also, `alphasort` is not a good solution as it would order devices as follows:

```
/dev/video0
/dev/video1
/dev/video10
/dev/video2
...
/dev/video9
```

**Comment by kkartaltepe** [comment]
> Also, alphasort is not a good solution as it would order devices as follows:

It seems a decent solution to enable building on freebsd without introducing a large amount of platform specific code. You could consider leaving the results unsorted on freebsd as well perhaps the devices will simply be in the proper order for most users.

**Comment by WizardCM** [comment]
From CI:
```
clang-format-12 is /usr/bin/clang-format-12
=================================
Files were not formatted properly
plugins/linux-v4l2/v4l2-output.c
```

**Comment by obiwac** [comment]
> You could consider leaving the results unsorted on freebsd as well perhaps the devices will simply be in the proper order for most users.

Can confirm this is not the case, unfortunately :(

Something like this could always be used:

```c
static int devsort(const struct dirent **_a, const struct dirent **_b) {
	int a = atoi((*_a)->d_name + 5);
	int b = atoi((*_b)->d_name + 5);

	return a - b;
}
```

This would eliminate the need for any platform specific code whatsoever, as it works both on Linux & FreeBSD. (Strings aren't sorted if not numeric.)

> From CI:
> 
> ```
> clang-format-12 is /usr/bin/clang-format-12
> =================================
> Files were not formatted properly
> plugins/linux-v4l2/v4l2-output.c
> ```

That's my bad, still figuring out how to use this `clang-format` stuff.

**Comment by obiwac** [comment]
Implemented that function I mentioned in my previous comment. Is it better like that or should I just choose between `alphasort` on FreeBSD and `versionsort` on Linux?

**Comment by obiwac** [comment]
Sure, I was operating under that assumption initially, and from that point of view I don't think anything bad could happen.

Regardless, I'll replace it with simple `alphasort`/`versionsort` depending on platform in a bit.

**Comment by obiwac** [comment]
Done

File: plugins/linux-v4l2/v4l2-output.c [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #define _GNU_SOURCE
 
+#include <ctype.h>
```
Comment (kkartaltepe) [inline plugins/linux-v4l2/v4l2-output.c:3]: I dont think this is needed anymore?

Comment (obiwac) [reply plugins/linux-v4l2/v4l2-output.c:3]: Indeed, my bad

**Comment by obiwac** [comment]
Yup, of course

**Comment by obiwac** [comment]
There we go!

**Review by kkartaltepe — APPROVED** [review]
You squashed the to_strig fix into the v4l2 fix if you could split that back out or make note in the commit desc.

**Comment by obiwac** [comment]
There we go.

**Pull request merged**